scene/gui/scroll_container.cpp

```diff
@@ -55,21 +53,23 @@ Size2 ScrollContainer::get_minimum_size() const {
 		largest_child_min_size = largest_child_min_size.max(child_min_size);
 	}
 
+	Size2 min_size;
+	Size2 size = get_size();
+
 	if (horizontal_scroll_mode == SCROLL_MODE_DISABLED) {
-		min_size.x = MAX(min_size.x, largest_child_min_size.x);
-	}
-	if (vertical_scroll_mode == SCROLL_MODE_DISABLED) {
-		min_size.y = MAX(min_size.y, largest_child_min_size.y);
+		min_size.x = largest_child_min_size.x;
+		bool v_scroll_show = vertical_scroll_mode == SCROLL_MODE_SHOW_ALWAYS || (vertical_scroll_mode == SCROLL_MODE_AUTO && largest_child_min_size.y > size.y);
```

Comment:
I'm not sure if using current size in minimum size calculations is a good idea, but it doesn't seem to break anything.

Reply:
The way this ends up working is that `size.x` influences `min_size.y` and vice versa. This should make it so that minimum size for each axis can't end up depending on itself.